### PR TITLE
Azure App Service - added support for listing publishing credentials

### PR DESCRIPTION
**Pull request by vincentdass**
PR for the feature request: #8505

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by vincentdass** [comment]
@yugangw-msft , Can you please share the steps for adding tests?

**Comment by tjprescott** [comment]
@vincentdass https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md

**Comment by tjprescott** [comment]
Closes #8505.

**Comment by vincentdass** [comment]
@yugangw-msft , @tjprescott , added test.
Please review the same.

File: src/command_modules/azure-cli-appservice/HISTORY.rst [hunk 1]
```diff
@@ -4,6 +4,10 @@ Release History
 ===============
 * functionapp: az functionapp devops-build, new command created
 
+0.2.14
+++++++
+* webapp, functionapp: az webapp/functionapp deployment list-publishing-credentials, get the Kudu (scm) url and its credentials
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/HISTORY.rst]: Can you move this change log to top? and then I will merge

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 2]
```diff
@@ -277,6 +277,14 @@
     crafted: true
 """
 
+helps['functionapp deployment list-publishing-credentials'] = """
+type: command
+short-summary: Get the details for available function app publishing credentials.
+examples:
+  - name: Get the details for available function app deployment publishing credentials.
+    text: az functionapp deployment list-publishing-credentials --name MyFunctionApp   --resource-group MyResourceGroup
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: you need to add a `crafted:true` here too

File: src/command_modules/azure-cli-appservice/HISTORY.rst (hunk 1 above)
Comment (vincentdass) [reply src/command_modules/azure-cli-appservice/HISTORY.rst]: changed.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py (hunk 2 above)
Comment (vincentdass) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: added.

**Comment by vincentdass** [comment]
@yugangw-msft , addressed review comments.

File: src/command_modules/azure-cli-appservice/HISTORY.rst [hunk 3]
```diff
@@ -3,10 +3,10 @@
 Release History
 ===============
 
-* Remove erroneous print statement for `az webapp auth update`
-
 0.2.14
 ++++++
+* webapp, functionapp: az webapp/functionapp deployment list-publishing-credentials, get the Kudu (scm) url and its credentials
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/HISTORY.rst]: you should move this to be above the `0.2.14`. 0.2.14 has been released.

Comment (vincentdass) [reply src/command_modules/azure-cli-appservice/HISTORY.rst]: done.

File: src/command_modules/azure-cli-appservice/HISTORY.rst [hunk 4]
```diff
@@ -2,11 +2,11 @@
 
 Release History
 ===============
-
-* Remove erroneous print statement for `az webapp auth update`
+* webapp, functionapp: az webapp/functionapp deployment list-publishing-credentials, get the Kudu (scm) url and its credentials
 
 0.2.14
 ++++++
+* Remove erroneous print statement for `az webapp auth update`
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/HISTORY.rst]: why you moved anot yet released change to be a released one?

Comment (vincentdass) [reply src/command_modules/azure-cli-appservice/HISTORY.rst]: changed.

**Pull request merged**